common/src/main/java/org/corfudb/common/util/URLUtils.java

```diff
@@ -66,6 +66,11 @@ public static String getVersionFormattedEndpointURL(String address) {
      */
     public static String getVersionFormattedHostAddress(String host) {
 
+        // In LR-SaaS, host could be null.
+        if (host == null || host.isEmpty()){
+            return "";
+        }
+
```

Comment:
Since this breaks the contract for the callers (i.e., expect a valid formatted address), I think you should push this concern to the caller instead of transforming it to an empty String.

Reply:
Done, Throwing an `IllegalArgumentException` now. Thank you!

Reply:
It's not clear what this method is suppose to do, but it seems buggy. For example, you can pass "localhost]" and it wouldn't fail despite the host being malformed. 

I think you should rename the method to `validateAndFormat` and do an IPv4, or an IPv6 validation. 


Also, did you submit this patch to fix an existing bug, or is it just to improve readability? If it's the prior, then you need to change the calling code paths to make sure that only non-null addresses are being supplied to the method.

Reply:
The purpose of this method was to format IPv6 addresses mainly, I gave a generic name since we call this in many places, and this can be extended in the future for any of the IPv4 formattings as well. 

It is not intended to validate the IP addresses. `"localhost]"` should not fail. In some of our universe tests using containers, we have Corfu Server alias addresses that are mapped to virtual addresses. And these alias addresses need not be valid IPv4 or IPv6 addresses. This method should work even for such addresses.

Agreed on changing the calling code everywhere. On top of my mind, I can't think of any existing place, I will fix it as we face them. For now, the current method passed all the GitHub Actions.